### PR TITLE
HIR docs: mention how to resolve method paths

### PR DESCRIPTION
**Pull request by RalfJung**

### CODE REVIEW

**Comment by RalfJung** [comment]
@oli-obk I hope I got this right.

File: src/librustc/hir/mod.rs [hunk 1]
```diff
@@ -1616,6 +1616,11 @@ pub enum ExprKind {
     /// and the remaining elements are the rest of the arguments.
     /// Thus, `x.foo::<Bar, Baz>(a, b, c, d)` is represented as
     /// `ExprKind::MethodCall(PathSegment { foo, [Bar, Baz] }, [x, a, b, c, d])`.
+    ///
+    /// To resolve the called method to a `DefId`, call [`type_dependent_def_id`] with
+    /// the `hir_id` of the `MethodCall` node itself.
```
Comment (RalfJung) [inline src/librustc/hir/mod.rs]: This is based on my experiments in https://github.com/rust-lang/rust/pull/66044#discussion_r341932961. I hope this is right.

Comment (davidtwco) [reply src/librustc/hir/mod.rs]: This seems reasonable to me, but I can't speak to if it's the right way to do this.

**Review by davidtwco — APPROVED** [review]
LGTM, r=me unless you want someone else to take a look.

File: src/librustc/hir/mod.rs (hunk 1 above)
Comment (RalfJung) [reply src/librustc/hir/mod.rs]: Hm, who would know such things? @Centril @petrochenkov could you have a look?

File: src/librustc/hir/mod.rs [hunk 2]
```diff
@@ -1616,6 +1616,11 @@ pub enum ExprKind {
...
+    ///
+    /// [`qpath_res`]: ../ty/struct.TypeckTables.html#method.type_dependent_def_id
```
Comment (oli-obk) [inline src/librustc/hir/mod.rs]:
```suggestion
    /// [`type_dependent_def_id`]: ../ty/struct.TypeckTables.html#method.type_dependent_def_id
```

**Comment by oli-obk** [comment]
yea these are the right functions to call. It's good to have this documented, I've looked for this function a few times and kept rediscovering it and forgetting it again

File: src/librustc/hir/mod.rs (hunk 2 above)
Comment (RalfJung) [reply src/librustc/hir/mod.rs]: Ah, why didn't CI catch this? I thought we check markdown for errors?

**Pull request merged**